lib/core/src/Cardano/Wallet/Registry.hs

```diff
@@ -190,44 +200,51 @@ defaultWorkerAfter tr = traceWith tr . \case
         Just UserInterrupt -> MsgUserInterrupt
         _ -> MsgUnhandledException $ pretty $ show e
 
--- | Create a new worker for a given key. Workers maintain an acquired resource.
--- They expect a task as argument and will terminate as soon as their task
--- is over; so in practice, we provide a never-ending task that keeps the worker
--- alive forever.
+-- | Register a new worker for a given key.
+--
+-- A worker maintains an acquired resource. It expects a task as an argument
+-- and will terminate as soon as its task is over. In practice, we provide a
+-- never-ending task that keeps the worker alive forever.
 --
 -- Returns 'Nothing' if the worker fails to acquire the necessary resource or
--- terminate unexpectedly before entering its 'main' action.
+-- terminates unexpectedly before entering its 'main' action.
 --
--- >>> newWorker ctx k withDBLayer restoreWallet
--- worker<thread#1234>
-newWorker
+register
     :: forall resource ctx key msg.
-        ( key ~ WorkerKey ctx
+        ( Ord key
+        , key ~ WorkerKey ctx
         , msg ~ WorkerMsg ctx
         , HasLogger (WorkerLog key msg) ctx
         , HasWorkerCtx resource ctx
         )
     => ctx
     -> key
+    -> WorkerRegistry key resource
```

Comment:
Side-note, all function working with the registry takes the registry as first argument. It'd be more consistent to also do this for this new function.

Reply:
@KtorZ wrote:
> Side-note, all function working with the registry takes the registry as first argument. It'd be more consistent to also do this for this new function.

Fixed in commit af7722992e91af4d8e8ccf5cc8826f8748cc6949.